### PR TITLE
Issue 105: Resource list not showing up with MySQL 5.7

### PR DESCRIPTION
**Pull request by veggiematts**
This patch fixes resource list for MySQL 5.7.

See http://stackoverflow.com/questions/34115174/i-am-getting-an-error-in-mysql-related-to-only-full-group-by-when-executing-a-qu for reference.

Fixes issue #105

### CODE REVIEW

**Comment by tuxayo** [comment]
Also works with `MariaDB 10.1.16` + `PHP 7.0.10` + Arch Linux
Tested to check if it doesn't break compatibility.

**Comment by tuxayo** [comment]
No issues with this patch when doing three installs on different machines with MySQL5.7 + Ubuntu 16.04 + PHP 7

**Comment by veggiematts** [comment]
Can someone have a second look on this? It's a really small patch :)

File: resources/admin/classes/domain/Resource.php [hunk 1]
```diff
@@ -1366,7 +1366,7 @@ public function getAlphabeticalList() {
 		$alphArray = array();
 		$result = $this->db->query("SELECT DISTINCT UPPER(SUBSTR(TRIM(LEADING 'The ' FROM titleText),1,1)) letter, COUNT(SUBSTR(TRIM(LEADING 'The ' FROM titleText),1,1)) letter_count
 								FROM Resource R
-								GROUP BY SUBSTR(TRIM(LEADING 'The ' FROM titleText),1,1)
+								GROUP BY SUBSTR(TRIM(LEADING 'The ' FROM titleText),1,1), titleText
```
Comment (remocrevo) [inline resources/admin/classes/domain/Resource.php]: I think we actually want the GROUP BY clause to match the first part of the SELECT clause. I'm guessing that would prevent the warning. I would suggest testing this:

GROUP BY UPPER(SUBSTR(TRIM(LEADING 'The ' FROM titleText),1,1))

Comment (veggiematts) [inline resources/admin/classes/domain/Resource.php]: Your solution worked on my instance.

I would go for any solution, but I'm not sure which one is the best, performance-wise (or if they are equivalent)

Comment (benheet) [inline resources/admin/classes/domain/Resource.php]: Looks like a decision needs to be made here between two different solutions. @remocrevo or @veggiematts do either of you have the time to modify this to use the solution that prevents the warning? If not would it be acceptable to accept this as is and then add an issue for making this modification later?

**Comment by benheet** [comment]
Let's go with Remington's proposed solution on this. @veggiematts can you update this to use Remington's proposed solution which you successfully tested and then we'll merge it in?

**Comment by veggiematts** [comment]
Done.

**Pull request merged**